### PR TITLE
[FIX] mail, web: activities, 'dl-horizontal' class is now discontinued

### PR DESCRIPTION
**Pull request by stefanorigano**
Prior to this commit the 'o_Activity_details' element was using a
discontinued bootstrap class, causing the layout to brake.

task#2731819.

required by: https://github.com/odoo/odoo/pull/82590

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/activity/activity.xml [hunk 1]
```diff
@@ -114,10 +120,10 @@
                     </t>
 
                     <t t-if="activityView.activity.canWrite">
-                        <div name="tools" class="o_Activity_tools">
+                        <div name="tools" class="o_Activity_tools d-flex">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:123]: I think there is a problem with the split in different PR here, this and everything below should not be part of the `dl` changes.

File: addons/mail/static/src/components/activity/activity.xml [hunk 2]
```diff
@@ -47,51 +47,57 @@
                                 <t t-esc="activityView.assignedUserText"/>
                             </div>
                         </t>
-                        <a class="o_Activity_detailsButton btn btn-link" t-on-click="activityView.onClickDetailsButton" role="button">
+                        <a
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:50]: not related to `dl` change either?

File: addons/mail/static/src/components/activity/activity.scss [hunk 3]
```diff
@@ -167,16 +159,6 @@ a:not([href]):not([tabindex]).o_Activity_detailsButton {
...
 
-.o_Activity_toolButton {
-    opacity: 0.5;
-    color: gray('500');
-
-    &:hover {
-        opacity: 1;
-        color: gray('600');
-    }
-}
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.scss:178]: all of this not related to `dl` changes either?

File: addons/web/static/src/legacy/scss/utils.scss [hunk 4]
```diff
@@ -195,8 +195,13 @@
...
     &, &:hover, &:focus, &:active, &.active {
-        border-color: transparent;
-        background-color: transparent;
+        border-color: transparent !important;
+        background-color: transparent !important;
+    }
+
+    &:hover:active:focus {
+        box-shadow: none;
+        outline: none;
```
Comment (seb-odoo) [inline addons/web/static/src/legacy/scss/utils.scss:204]: also not related to `dl` changes?

File: addons/mail/static/src/components/activity/activity.xml [hunk 5]
```diff
@@ -47,51 +47,57 @@
...
                     </div>
 
                     <t t-if="activityView.areDetailsVisible">
                         <div class="o_Activity_details">
-                            <dl class="dl-horizontal">
-                                <t t-if="activityView.activity.type">
-                                    <dt>Activity type</dt>
-                                    <dd class="o_Activity_type">
+                            <div class="d-table table table-sm mt-2 mb-3">
+                                <div class="d-table-row" t-if="activityView.activity.type">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:62]: small detail: our guideline is to have `t-if` attribute before `class`

File: addons/mail/static/src/components/activity/activity.xml [hunk 6]
```diff
@@ -47,51 +47,57 @@
...
-                                    <dd class="o_Activity_detailsCreation">
-                                        <t t-esc="activityView.formattedCreateDatetime"/>
-                                        <img class="o_Activity_detailsUserAvatar o_Activity_detailsCreatorAvatar" t-attf-src="/web/image/res.users/{{ activityView.activity.creator.id }}/avatar_128" t-att-title="activityView.activity.creator.nameOrDisplayName" t-att-alt="activityView.activity.creator.nameOrDisplayName"/>
-                                        <span class="o_Activity_detailsCreator">
+                                    </span>
+                                </div>
+                                <div class="d-table-row" t-if="activityView.activity.creator">
+                                    <b class="d-table-cell text-right py-1 px-4">Created</b>
+                                    <span class="o_Activity_detailsCreation d-table-cell py-1 pr-4">
+                                        <t t-esc="activityView.formattedCreateDatetime"/>, by&#160;
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:71]: what is `&#160;`?

File: addons/mail/static/src/components/activity/activity.xml [hunk 7]
```diff
@@ -47,51 +47,57 @@
...
+                                        <img class="o_Activity_detailsUserAvatar o_Activity_detailsCreatorAvatar rounded-circle" t-attf-src="/web/image/res.users/{{ activityView.activity.creator.id }}/avatar_128" t-att-title="activityView.activity.creator.nameOrDisplayName" t-att-alt="activityView.activity.creator.nameOrDisplayName"/>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:72]: adding `rounded-circle` does not look related to `dl` changes?

File: addons/mail/static/src/components/activity/activity.xml [hunk 8]
```diff
@@ -47,51 +47,57 @@
...
+                                    <span class="o_Activity_detailsDueDate d-table-cell py-1 pr-4">
+                                        <span class="o_Activity_deadlineDateText"
+                                            t-att-class="{
+                                                'o-default': activityView.activity.state === 'default',
+                                                'o-overdue': activityView.activity.state === 'overdue',
+                                                'o-planned': activityView.activity.state === 'planned',
+                                                'o-today': activityView.activity.state === 'today',
+                                            }"
+                                        >
+                                            <t t-esc="activityView.formattedDeadlineDate"/>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:96]: Why is github marking a lot of these lines as changed?

I don't see any change in them, but maybe it will show something different in editor?

File: addons/mail/static/src/components/activity/activity.xml (hunk 6 above)
Comment (stefanorigano) [reply addons/mail/static/src/components/activity/activity.xml:71]: Uops, 'inline-margin' should be used instead. Thanks 👍

File: addons/mail/static/src/components/activity/activity.xml (hunk 8 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/activity/activity.xml:96]: Ok I get it, there is a partial indentation change -> the blocks that had `t-if` don't have indent change because you re-used the `t-if` level to add the `div`, but those that didn't have the condition now have an extra div.

File: addons/mail/static/src/components/activity/activity.xml [hunk 9]
```diff
@@ -47,51 +47,57 @@
...
+                                    <b class="d-table-cell text-right py-1 px-4">Created</b>
+                                    <span class="o_Activity_detailsCreation d-table-cell py-1 pr-4">
+                                        <t t-esc="activityView.formattedCreateDatetime"/>, by&#160;
+                                        <img class="o_Activity_detailsUserAvatar o_Activity_detailsCreatorAvatar rounded-circle" t-attf-src="/web/image/res.users/{{ activityView.activity.creator.id }}/avatar_128" t-att-title="activityView.activity.creator.nameOrDisplayName" t-att-alt="activityView.activity.creator.nameOrDisplayName"/>
+                                        <b class="o_Activity_detailsCreator">
                                             <t t-esc="activityView.activity.creator.nameOrDisplayName"/>
+                                        </b>
+                                    </span>
+                                </div>
+                                <div class="d-table-row" t-if="activityView.activity.assignee">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:78]: just to make sure: the `t-if` order guideline applies to all of them in the diff of course

File: addons/mail/static/src/components/activity/activity.xml [hunk 10]
```diff
@@ -47,51 +47,57 @@
...
-                                </t>
-                                <t t-if="activityView.activity.creator">
-                                    <dt>Created</dt>
-                                    <dd class="o_Activity_detailsCreation">
-                                        <t t-esc="activityView.formattedCreateDatetime"/>
-                                        <img class="o_Activity_detailsUserAvatar o_Activity_detailsCreatorAvatar" t-attf-src="/web/image/res.users/{{ activityView.activity.creator.id }}/avatar_128" t-att-title="activityView.activity.creator.nameOrDisplayName" t-att-alt="activityView.activity.creator.nameOrDisplayName"/>
-                                        <span class="o_Activity_detailsCreator">
+                                    </span>
+                                </div>
+                                <div class="d-table-row" t-if="activityView.activity.creator">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:68]: also `t-if` order

File: addons/mail/static/src/components/activity/activity.xml (hunk 8 above)
Comment (stefanorigano) [reply addons/mail/static/src/components/activity/activity.xml:96]: > but maybe it will show something different in editor?

Nope :/
Somehow "fixed" ( I just reverted the line and amended the commit, not pushed yet)

File: addons/mail/static/src/components/activity/activity.xml (hunk 1 above)
Comment (stefanorigano) [reply addons/mail/static/src/components/activity/activity.xml:123]: Confirmed, thanks

**Comment by seb-odoo** [comment]
Thank you for the update, now the PR is clean and easy to see what is going on.

But now I still don't understand why we are trying to use a table layout instead of flex or grid.

The previous code was much better on mobile due to going to multiple lines:

now
image

before
image

**Comment by seb-odoo** [comment]
> `.dl-horizontal has been dropped. Instead, use .row on <dl> and use grid column classes (or mixins) on its <dt> and <dd> children.`

The deprecation notice also suggests that grid should be used, not table. `dl` and `dd` are still semantically correct, we just need to change their style.

**Comment by stefanorigano** [comment]
Hello @seb-odoo ,
thanks, I should have handled mobile, fixed.

I restored the description list tags  (`dl`, `dt`..), changing them was indeed unnecessary. 
> .dl-horizontal has been dropped. Instead, use .row on `<dl>` and use grid column classes (or mixins) ...
>> The deprecation notice also suggests that grid should be used, not table.

"grid **column classes**", so  `col-3` etc.. (bts implementation example [here](https://getbootstrap.com/docs/4.3/content/typography/#description-list-alignment)).
The problem with this implementation is that is quite strict and easily brake on translations.
Eg: maybe `col-2` is perfect in English, but it may truncates labels in Portuguese and despite having plenty of available space :/

`display: grid` is definitively an option, but we'll have to wait for BTS5 for that... I really don't want to define a temporary entire grid system just for this ^^

Moreover, the final appearance will change for sure, for v16 the idea is to re-use the same design as the technical messages (check screenshoot in attach.) .
The aim of this commit is just to fix the issue and to allow me to review the SCSS in https://github.com/odoo/odoo/pull/82590 .
We'll fine tune the design (possibly with a generic class) when implementing the new design 👍 
image

File: addons/mail/static/src/components/activity/activity.xml [hunk 11]
```diff
@@ -54,44 +54,46 @@
...
+                                </dl>
+                                <dl t-if="activityView.activity.assignee" class="d-md-table-row mb-3">
+                                    <dt class="d-md-table-cell text-md-right m-0 py-md-1 px-md-4">Assigned to</dt>
+                                    <dd class="o_Activity_detailsAssignation d-md-table-cell py-md-1 pr-4">
                                         <img class="o_Activity_detailsUserAvatar o_Activity_detailsAssignationUserAvatar" t-attf-src="/web/image/res.users/{{ activityView.activity.assignee.id }}/avatar_128" t-att-title="activityView.activity.assignee.nameOrDisplayName" t-att-alt="activityView.activity.assignee.nameOrDisplayName"/>
-                                        <t t-esc="activityView.activity.assignee.nameOrDisplayName"/>
+                                        <b t-esc="activityView.activity.assignee.nameOrDisplayName"/>
+                                    </dd>
+                                </dl>
+                                <dl class="d-md-table-row">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/activity/activity.xml:81]: I was hoping that keeping `dt`/`dd` would prevent these big diff indent change, but if you have to introduce one `dl` per line anyway, I think it makes no sense to keep it, and we are better off just using `div` everywhere.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

Hopefully we can replace table display by grid layout at some point.
And also avoid repetition of group of classes.

**Comment by stefanorigano** [comment]
hello @odoo/rd-framework-js , little fix for mail
